### PR TITLE
FALCON-2170 Umbrella jira for bugs in EntitySLAMonitoring and Backlog…

### PR DESCRIPTION
**Pull request by PraveenAdlakha**
…Emitter service

### CODE REVIEW

**Comment by PraveenAdlakha** [comment]
Hi @sandeepSamudrala , @pallavi-rao 

As suggested and discussed this is the pull request for the umbrella jira which have all the fixes for Backlog and EntitySLAMonitoring.

Also, this pull request includes work from me and @sandeepSamudrala  the subtasks have the rightful owner.

Please review.

Thanks
Praveen

File: messaging/src/main/java/org/apache/falcon/messaging/JMSMessageConsumer.java [hunk 1]
```diff
@@ -92,8 +92,7 @@ public void startSubscriber() throws FalconException {
 
             topicSession = (TopicSession) connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
             Topic destination = topicSession.createTopic(topicName);
-            topicSubscriber = topicSession.createDurableSubscriber(destination, FALCON_CLIENT_ID,
-                    WorkflowNameBuilder.WorkflowName.getJMSFalconSelector(), false);
+            topicSubscriber = topicSession.createDurableSubscriber(destination, FALCON_CLIENT_ID);
```
Comment (pallavi-rao) [inline messaging/src/main/java/org/apache/falcon/messaging/JMSMessageConsumer.java]: Why this change?

File: oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java [hunk 2]
```diff
@@ -959,13 +958,6 @@ private void performAction(String cluster, JobAction action, CoordinatorAction c
             status = Status.KILLED.name();
             break;
 
-        case IGNORE:
```
Comment (pallavi-rao) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: This change is reversing the patch to ensure manually killed instances aren't retried.

File: oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java [hunk 3]
```diff
@@ -1113,12 +1105,10 @@ private String mapActionStatus(String status) {
         } else if (CoordinatorAction.Status.WAITING.toString().equals(status)
             || CoordinatorAction.Status.SUBMITTED.toString().equals(status)) {
             return InstancesResult.WorkflowStatus.WAITING.name();
-        } else if (CoordinatorAction.Status.KILLED.toString().equals(status)) {
-            return InstancesResult.WorkflowStatus.KILLED.name();
         } else if (CoordinatorAction.Status.IGNORED.toString().equals(status)) {
-            return InstancesResult.WorkflowStatus.KILLED_OR_IGNORED.name();
+            return InstancesResult.WorkflowStatus.KILLED.name();
```
Comment (pallavi-rao) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: This change is reversing a patch to ensure manually killed instances aren't retried.

File: oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java [hunk 4]
```diff
@@ -1113,12 +1105,10 @@ private String mapActionStatus(String status) {
...
         } else if (CoordinatorAction.Status.TIMEDOUT.toString().equals(status)) {
-            return InstancesResult.WorkflowStatus.TIMEDOUT.name();
+            return InstancesResult.WorkflowStatus.FAILED.name();
```
Comment (pallavi-rao) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: This change is reversing a patch to ensure manually killed instances aren't retried.

File: prism/src/main/java/org/apache/falcon/plugin/GraphiteNotificationPlugin.java [hunk 5]
```diff
@@ -50,14 +50,12 @@ public void monitor(ResourceMessage message) {
...
                     ? message.getDimensions().get("entityName") :message.getDimensions().get("entity-name");
             String prefix = StartupProperties.get().getProperty("falcon.graphite.prefix");
-            String separator = ".";
             LOG.debug("message:" + message.getAction());
             if (entityType.equalsIgnoreCase(EntityType.PROCESS.name())
                     && ConfigurationStore.get().get(EntityType.PROCESS, entityName) != null) {
                 Entity entity = ConfigurationStore.get().get(EntityType.PROCESS, entityName);
                 Process process = (Process) entity;
                 String pipeline =  StringUtils.isNotBlank(process.getPipelines()) ? process.getPipelines() : "default";
-                pipeline = pipeline.replaceAll(" ", "-");
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/plugin/GraphiteNotificationPlugin.java]: This is change is essential to ensure all pipeline tags are used. isn't it?

File: prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java [hunk 6]
```diff
@@ -205,7 +261,7 @@ public void onWait(WorkflowExecutionContext context) throws FalconException {
 
         @Override
         public void run() {
-            LOG.debug("Starting periodic check for backlog");
+            LOG.trace("BacklogMetricEmitter running for entities");
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java]: Intentional change. Keep the originial.

File: oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java (hunk 3 above)
Comment (PraveenAdlakha) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: yeah will make the required change around this.

File: messaging/src/main/java/org/apache/falcon/messaging/JMSMessageConsumer.java (hunk 1 above)
Comment (PraveenAdlakha) [inline messaging/src/main/java/org/apache/falcon/messaging/JMSMessageConsumer.java]: its not required will remove it .

File: prism/src/main/java/org/apache/falcon/plugin/GraphiteNotificationPlugin.java (hunk 5 above)
Comment (PraveenAdlakha) [inline prism/src/main/java/org/apache/falcon/plugin/GraphiteNotificationPlugin.java]: yes its reversing the patch will correct it.
